### ISSUE TITLE
[BUG] rancher-system-agent keeps showing error on a new node in an upgraded cluster

### ISSUE THREAD

**Issue by FrankYang0529** [harvester/harvester#2892]
**Describe the bug**
Upgrade a cluster from v1.0.3 to v1.1.0-rc2. Delete an agent node after the upgrade is done.
When we re-join a new node (with the same hostname) to the cluster, the `rancher-system-agent` keeps showing errors.
```
Oct 07 03:35:06 harvester-node-3 rancher-system-agent[2009]: W1007 03:35:06.728098    2009 reflector.go:443] pkg/
mod/github.com/rancher/client-go@v1.24.0-rancher1/tools/cache/reflector.go:168: watch of *v1.Secret ended with: a
n error on the server ("unable to decode an event from the watch stream: stream error: stream ID 5; INTERNAL_ERRO
R; received from peer") has prevented the request from succeeding
```

After restarting the `rancher-system-agent` on the new node, the error message changes.
```
Oct 07 04:05:55 harvester-node-3 rancher-system-agent[31948]: time="2022-10-07T04:05:55Z" level=info msg="Startin
g remote watch of plans"
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]: panic: error while connecting to Kubernetes cluster: Get "https://192.168.3.131/version": dial tcp 192.168.3.131:443: connect: connection timed out
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]: goroutine 38 [running]:
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]: github.com/rancher/system-agent/pkg/k8splan.(*watcher).start(0xc00010a200, {0x18bd5c0?, 0xc0001ee640})
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]:         /go/src/github.com/rancher/system-agent/pkg/k8splan/watcher.go:102 +0xa13
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]: created by github.com/rancher/system-agent/pkg/k8splan.Watch
Oct 07 04:08:05 harvester-node-3 rancher-system-agent[31948]:         /go/src/github.com/rancher/system-agent/pkg/k8splan/watcher.go:63 +0x155
Oct 07 04:08:05 harvester-node-3 systemd[1]: rancher-system-agent.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Oct 07 04:08:05 harvester-node-3 systemd[1]: rancher-system-agent.service: Failed with result 'exit-code'.
Oct 07 04:08:10 harvester-node-3 systemd[1]: rancher-system-agent.service: Scheduled restart job, restart counter is at 4.
Oct 07 04:08:10 harvester-node-3 systemd[1]: Stopped Rancher System Agent.
```

Also, the `cattle-system/apply-system-agent-upgrader-on-xxx` job can't complete.
```
[ERROR]  Error while connecting to Rancher to verify CA necessity. Sleeping for 5 seconds and trying again.
```

**To Reproduce**
1. Start a 4-node v1.0.3 harvester cluster.
2. Apply a new version and update the cluster.
```
cat <<EOF | kubectl apply -f -
apiVersion: harvesterhci.io/v1beta1
kind: Version
metadata:
  name: v1.1.0-rc2
  namespace: harvester-system
spec:
  isoURL: "http://192.168.0.181:8000/harvester-master-amd64.iso"
EOF
```
3. After upgrade successfully, remove the agent node and rejoin it again.

**Expected behavior**
The agent node can join successfully.

**Support bundle**
[agent-node-rancher-system-agent-log.txt](https://github.com/harvester/harvester/files/9730634/agent-node-rancher-system-agent-log.txt)

**Environment**
 - Harvester ISO version: v1.0.3 to v1.1.0-rc2
 - Underlying Infrastructure: KVM

**Comment by FrankYang0529** [comment]
This issue also needs fixes in https://github.com/harvester/harvester/issues/2893 and https://github.com/harvester/harvester/issues/2734. Other fixes will be included in https://github.com/harvester/harvester/pull/2918.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-0a9538a1-head` (10/14). Close this issue.

### Result
Verified after upgrading from v1.0.3 to v1.1.0 master release, we can rejoin the management node and agent node back correctly. 

1. Successfully re-join the management node after upgrade 
  ![image](https://user-images.githubusercontent.com/29251855/196159969-9c8acb11-b9fe-4501-94d5-74545579ef4d.png)

2. Successfully re-join the agent node after upgrade 
  ![image](https://user-images.githubusercontent.com/29251855/196164183-20c4498a-c4c6-4033-a527-d3c8ff84f8aa.png)


### Test Information
* Test Environment: 4 nodes harvester on local kvm machine
* Harvester version: `master-0a9538a1-head` (10/14)

### Verify Steps
1. Create a 4-node v1.0.3 cluster.
2. Upgrade the master branch:
 - Check the spec content in `provisioning.cattle.io/v1/clusters -> fleet-local`
![image](https://user-images.githubusercontent.com/29251855/196139161-7b6e6e84-692d-4f4f-a978-62fc50f64f06.png)
 - Check the iface content in `helm.cattle.io/v1/helmchartconfigs -> rke2-canal`
![image](https://user-images.githubusercontent.com/29251855/196141139-e4ff668e-287a-4722-865d-a7c0f145c862.png)
  ```
   spec:                                                                                                                                                                                      │
  │   valuesContent: |-                                                                                                                                                                        │
  │     flannel:                                                                                                                                                                               │
  │       iface: "" 
  ```

3. Remove the agent node and 1 management node.
   - Remove agent node (node 4)
  ![image](https://user-images.githubusercontent.com/29251855/196138324-83b64d50-0236-4110-86c2-551ae046a406.png)
  ![image](https://user-images.githubusercontent.com/29251855/196138906-e77de8f0-d61d-4a91-82b2-e36cbbb6462a.png)
  
  - Remove management node (node 3)
  ![image](https://user-images.githubusercontent.com/29251855/196139707-ce8cc444-21fb-4bba-a040-9b09e0ae7fa9.png)
  ![image](https://user-images.githubusercontent.com/29251855/196139906-ff92320f-f5f7-4fb5-af20-5cf0f3fe36b9.png)
  ![image](https://user-images.githubusercontent.com/29251855/196147230-7c0b1835-c219-41b2-b0b8-32bae9962f27.png)

4. After the node is removed, provision a new node.
5. Check the node can join the cluster and can be promoted as a management node.
6. After we have 3 management nodes, provision a new node and check if it can join the cluster.
